OrbitQt/ConnectToStadiaWidget.h

```diff
@@ -58,22 +62,27 @@ class ConnectToStadiaWidget : public QWidget {
   void ReadyToDeploy();
   void Connect();
 
- protected:
-  bool eventFilter(QObject* obj, QEvent* event) override;
-
  private:
   std::unique_ptr<Ui::ConnectToStadiaWidget> ui_;
-  StadiaConnection* stadia_connection_ = nullptr;
   OrbitGgp::InstanceItemModel instance_model_;
-  QStateMachine state_machine_;
+  SshConnectionArtifacts* ssh_connection_artifacts_ = nullptr;
+  std::optional<OrbitGgp::Instance> selected_instance_;
+  std::unique_ptr<ServiceDeployManager> service_deploy_manager_;
+  std::shared_ptr<grpc::Channel> grpc_channel_;
   QPointer<OrbitGgp::Client> ggp_client_ = nullptr;
   std::optional<QString> remembered_instance_id_;
 
+  // State Machine
+  QStateMachine state_machine_;
+  QState* s_instances_loading_ = nullptr;
+  QState* s_connected_ = nullptr;
```

Comment:
This is trying to basically follow the Qt way of doing this. With `state_machine_.addState(QState*)`, the state machine takes ownership of the QState* and if I am informed correctly, takes car of deleting the states. 

I am sure that using unique_ptr or an object instead of pointers is possible, but I am unsure what the correct and best way of doing this is, so nothing is deleted twice. I can take some time to figure this out, or if you or @beckerhe know already and have a minute to explain, that would be great.

Reply:
I think that's what we discussed OOB.